rust/experimental/query_engine/ottl-parser/src/scalar_expression.rs

```diff
@@ -24,6 +24,7 @@ pub(crate) fn parse_scalar_expression(
         Rule::string_literal => {
             ScalarExpression::Static(parse_standard_string_literal(scalar_rule))
         }
+        Rule::null_literal => ScalarExpression::Static(parse_standard_null_literal(scalar_rule)),
```

Comment:
As mentioned in [the other comment](https://github.com/open-telemetry/otel-arrow/pull/727/files#r2216538634) - this should look like:

```suggestion
        Rule::nil_literal => ScalarExpression::Static(parse_standard_null_literal(scalar_rule)),
```

This matches the OTTL `nil_literal` rule against the `null_literal` parsing offered by `parser-abstractions`, the result being `NullScalarExpression` to be used in query engine.